docker-swarm.yml

```diff
@@ -0,0 +1,43 @@
+version: '3'
+
+services:
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.2.1
+    volumes:
+      - ./elasticsearch/config/elasticsearch.yml:/usr/share/elasticsearch/config/elasticsearch.yml:ro
+    ports:
+      - "9200:9200"
+      - "9300:9300"
+    environment:
+      ES_JAVA_OPTS: "-Xmx256m -Xms256m"
+    networks:
+      - elk
+
+  logstash:
+    image: docker.elastic.co/logstash/logstash-oss:6.2.1
+    volumes:
+      - ./logstash/config/logstash.yml:/usr/share/logstash/config/logstash.yml:ro
```

Comment:
I believe that this will work on a single node cluster but will probably fail on a cluster with nodes > 1.

A solution here would be to define the equivalent of the configuration file as environment variables.

Reply:
Agree, our current approach of mounting files was meant to make it easy to fiddle with configurations, locally. It doesn't apply very well to a distributed infrastructure.

For the sake of consistency it would be nice to move everything to env vars, unfortunately only a subset of variables is supported.

Reply:
Yup, I wonder if the defaults configuration variables we ship in configuration are part of that supported subset.

Reply:
ES definitely, Kibana and Logstash possibly. Search Guard unlikely :)

Reply:
I was thinking about adding this setup in the master branch only to start with. We'll think about other setups after that :-)